object_tracing: remove tracing from ring_buffer

nashif opened the pull request:
The ring buffer was missing the needed hooks needed for object tracing.
Removed rather than fixing, ring_buffer is not a kernel object and does not require tracing support

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

MaureenHelm commented on lib/ring_buffer/ring_buffer.c:
```diff
@@ -20,6 +21,30 @@ struct ring_element {
+struct ring_buf   *_trace_list_sys_ring_buf;
+
+/*
+ * Complete initialization of statically defined semaphores.
```
s/semaphores/ring buffers/

nashif replied:
fixed

andrewboie commented on include/linker/common-ram.ld:
```diff
@@ -62,6 +62,13 @@
+	SECTION_DATA_PROLOGUE(_sys_ring_buf_area, (OPTIONAL), SUBALIGN(4))
+	{
+		_sys_ring_buf_list_start = .;
+		KEEP(*(SORT_BY_NAME("._sys_ring_buf.static.*")))
```
Did you test this?
There is currently no mechanism to put ring buffers in this section, see definition of SYS_RING_BUF_DECLARE_POW2 and SYS_RING_BUF_DECLARE_SIZE.

tyrantlizard77 reviewed (commented):
I'm a little hesistant about this patch for 2 reasons.

1) I see nothing that actually puts the ring buffers in the sys_ring_buf_area section

2) these ring buffers are NOT kernel objects. this is library code. I don't feel they belong in the kernel object tracing mechanism, which is why when I wrote this I did not include them.

nashif commented:
ok, I am fine with that, then we need to revert 0588da0b4a2b6407f134abeab5b59e397530fd3e or remove the incomplete support that was added almost 2 years ago.

nashif committed:
ring_buffer: remove broken object_tracing support

The library has incomplete tracing support. Given that we are not a true
kernel object, remove support instead of fixing it.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

lpereira reviewed (approved):
Always happy to see patches only removing code.
